packages/flutter_tools/test/general.shard/resident_runner_test.dart

```diff
@@ -174,7 +174,10 @@ void main() {
           flutterDevice,
         ],
         stayResident: false,
-        debuggingOptions: DebuggingOptions.enabled(BuildInfo.debug),
+        debuggingOptions: DebuggingOptions.enabled(
+          BuildInfo.debug,
+          serveObservatory: false,
```

Comment:
This causes all of these tests to fail to cover the new code that this PR adds. Instead of disabling Observatory in these tests, the expectation that the request will be made to the service should be added to the `FakeVmServiceHost` instances.